### ISSUE TITLE
registerEntity ignored when @Id is also defined

### ISSUE THREAD

**Issue by erikgollot** [javers/javers#142]
My problem is that I need to store the two versions of an Entity. So each one has a specific value for its @Id field. The real identity of my entities is the field businessIdentity. So I tried to register my entities with the following code
```java
EntityDefinition def = new EntityDefinition(PersistentEntity.class,"businessIdentity");
builder.registerEntity(def);
Javers javers = builder.build();
```

The problem is that Javers (I look with the debugger), always take my @Id field as the object identity.

Do I miss something ?

**Comment by bartoszwalacik** [comment]
registerEntity(EntityDefinition) should have priority over annotation scanner. You mean that now we have diffrent situation, so @Id annotation overriders registerEntity(EntityDefinition)?

**Comment by erikgollot** [comment]
Yes it seems that @id has the priority. I tried to had @ diffignore nzar @id and javers says that my object des not have any identity...even if previously i've used registerEntity.

**Comment by bartoszwalacik** [comment]
Right, I'll check this out

**Comment by erikgollot** [comment]
Thanks !!

Let me know

**Comment by bartoszwalacik** [comment]
well, we have a test showing that Id property name given in EntityDefinition overrides
@Id annotation

https://github.com/javers/javers/blob/b523a05b64b86631ab9fd01da4616f6518590087/javers-core/src/test/groovy/org/javers/core/metamodel/clazz/EntityFactoryIdTest.groovy
see def "should use custom id property when given"()

could you come up with a failing test, which shows that in your case you have the opposite behaviour?
it could be done in your fork of javers

**Comment by erikgollot** [comment]
Ok i will do that tomorrow...I'm in France and it 11:30 pm 


Envoyé depuis Yahoo Mail pour Android

De:"Bartosz Walacik" <notifications@github.com>
Date:Mer j Avr PM à 23:22
Objet:Re: [javers] registerEntity ignored when @Id is also defined (#142)

well, we have a test showing that Id property name given in EntityDefinition overrides
@Id annotation

https://github.com/javers/javers/blob/b523a05b64b86631ab9fd01da4616f6518590087/javers-core/src/test/groovy/org/javers/core/metamodel/clazz/EntityFactoryIdTest.groovy
see def "should use custom id property when given"()

could you come up with a failing test, which shows that in your case you have the opposite behaviour?
it could be done in your fork of javers

**Comment by erikgollot** [comment]
In the following clone of javers on my GitHub : erikgollot/javers
|   |
|   |  |   |   |   |   |   |
| erikgollot/javersjavers - JaVers is a tool for auditing object-oriented data |
|  |
| Afficher sur githu... | Aperçu par Yahoo |
|  |
|   |


You will find a test : org.javers.core.examples.ShouldUseRegisterAndNotIdObjectTest that fail.
In fact, it fail if I instantiate a child class of my base class ShouldUseRegisterAndNotIdObject. If the base class is not abstract and if I instantiate this class, no problem.So the problem comes with inheritance
      De : Bartosz Walacik <notifications@github.com>
 À : javers/javers <javers@noreply.github.com> 
Cc : erikgollot <erik_gollot@yahoo.fr> 
 Envoyé le : Mercredi 29 avril 2015 23h22
 Objet : Re: [javers] registerEntity ignored when @Id is also defined (#142)
   
well, we have a test showing that Id property name given in EntityDefinition overrides
@Id annotationhttps://github.com/javers/javers/blob/b523a05b64b86631ab9fd01da4616f6518590087/javers-core/src/test/groovy/org/javers/core/metamodel/clazz/EntityFactoryIdTest.groovy
see def "should use custom id property when given"()could you come up with a failing test, which shows that in your case you have the opposite behaviour?
it could be done in your fork of javers—

**Comment by bartoszwalacik** [comment]
@erikgollot, well, i cant see any commit at your fork. Where is your failing test?

**Comment by erikgollot** [comment]
Oh sorry, i probably forgot to push.

I've made a test that shows the problem when we've inheritance. It works if not.

Envoyé depuis Yahoo Mail pour Android

De:"Bartosz Walacik" <notifications@github.com>
Date:Ven j Mai PM à 22:23
Objet:Re: [javers] registerEntity ignored when @Id is also defined (#142)

@erikgollot, well, i cant see any commit at your fork. Where is your failing test?

**Comment by erikgollot** [comment]
I cannot push now, i dont have access to my computer. I will do it asap

Envoyé depuis Yahoo Mail pour Android

De:"Bartosz Walacik" <notifications@github.com>
Date:Ven j Mai PM à 22:23
Objet:Re: [javers] registerEntity ignored when @Id is also defined (#142)

@erikgollot, well, i cant see any commit at your fork. Where is your failing test?

**Comment by erikgollot** [comment]
You can have a look at https://github.com/erikgollot/javers/tree/check-register-definition-priority/javers-core/src/test/java/org/javers/core/examples
Class ShouldUseRegisterAndNotIdObjectTest.java

**Comment by bartoszwalacik** [comment]
ok, but why have you closed the issue?

**Comment by erikgollot** [comment]
I will check, I don't know why maybe a mistake

Envoyé depuis Yahoo Mail pour Android

De:"Bartosz Walacik" <notifications@github.com>
Date:Lun j Mai PM à 22:43
Objet:Re: [javers] registerEntity ignored when @Id is also defined (#142)

ok, but why have you closed the issue?

**Comment by bartoszwalacik** [comment]
ok, now I can see the issue

**Comment by bartoszwalacik** [comment]
fixed in vesrion 1.2.8
http://javers.org/documentation/features/#release-notes
